src/utils/work/seekWorkResponse.js

```diff
@@ -5,34 +5,43 @@ import fetchAPI from 'utils/fetchAPI';
 import unpackResult from 'utils/work/unpackResult';
 import WorkResponseError from 'utils/WorkResponseError';
 
+const throwResponseError = (response) => {
+  throw new Error(`Error ${response.status}: ${response.text}`, { cause: response });
+};
+
 const seekFromS3 = async (ETag, experimentId) => {
   const response = await fetchAPI(`/v1/workResults/${experimentId}/${ETag}`);
-  const { signedUrl } = await response.json();
 
-  if (!signedUrl) return null;
+  if (!response.ok) {
+    if (response.status === 404) return null;
+
+    throwResponseError(response);
+  }
+
+  const { signedUrl } = await response.json();
   const storageResp = await fetch(signedUrl);
+
   if (!storageResp.ok) {
-    return null;
+    throwResponseError(storageResp);
   }
 
   return unpackResult(storageResp);
 };
 
-const seekFromAPI = async (
+const dispatchWorkRequest = async (
   experimentId,
   body,
   timeout,
   ETag,
-  eventCallback,
   requestProps = {},
 ) => {
-  console.error('seek from api', body);
+  console.error('dispatching work request', body);
   const { default: connectionPromise } = await import('utils/socketConnection');
   const io = await connectionPromise;
 
   const timeoutDate = moment().add(timeout, 's').toISOString();
   const authJWT = await getAuthJWT();
-  const socketId = !eventCallback ? io.id : 'broadcast';
```

Comment:
Why was `broadcast` removed? Don't we use it for clustering?

Reply:
This is never used. If you take a look at the current call to `seekFromAPI` in `fetchWork` (https://github.com/hms-dbmi-cellenics/ui/blob/master/src/utils/work/fetchWork.js#L77), the function parameter is explicitly set to `null`.

Reply:
But that's only in `fetchGeneExpression`, right? In `fetchWork` we do use it https://github.com/hms-dbmi-cellenics/ui/blob/master/src/utils/work/fetchWork.js#L149

Reply:
Oh right. Right now `eventCallback` is only used when running celllSets clustering. I think this was meant to automatically update the cell sets automatically if we have multiple users looking at the same experiment in different browsers. This callback will automatically update the cell sets if clustering is changed in one of the browsers. 

If the socketId is passed as `broadcast` by the UI, the worker will send the response to the channel `${experimentId}-${body.name}` (https://github.com/hms-dbmi-cellenics/worker/blob/master/python/src/worker/response.py#L93-L99). Right now this is the channel that's listened to in the UI for the work request. However, we only open a listener for this channel when the UI dispatches `runCellSetsClsutering`. If the user never run cell sets clustering (e.g. jump straight from data management to data exploration), the listener will not be set up, and changes to the cell set by a user will not cause changes to the cell sets of other online users.

Reply:
To solve this issue, we can : 
1. Try to use other channels that we're currently always listening to.

The only socket channel we listen to all the time is `ExperimentUpdates-${currentExperimentId}` which is initialized in the content wrapper (https://github.com/hms-dbmi-cellenics/ui/blob/c9de20448ba52adcd9017f0098eec3fe319fad05/src/components/ContentWrapper.jsx#L118). Right now the callback for this socket handles updates for GEM2S and QC. We can make a switch in the `experimentUpdatesHandler` (https://github.com/hms-dbmi-cellenics/ui/blob/d025c80fd02ed497ca3bc30d694e33ceba9ac8ec/src/utils/experimentUpdatesHandler.js#L12) to also handle cell sets update.

Pros : 
- Cleaner, we don't have to add another socket to listen to

Cons : 
- More work in `experimentUpdatesHandler`, but it still falls in the scope of the handler's work.

2. Make the UI always listen to the 'broadcast' channel (e.g. `${experimentId}-${body.name}`) and handle the cellsets change there.

We can do this by initializing a socket to the channel and registering the callback to the listener of the broadcast channel.

Pros : 
- Dedicated channel to listen for updates. Cleaner binding.

Cons:
- Duplicated code (more channels to listen to)
- Adding a listener on a channel which users may rarely use, especially if they have found a good cluster to use.

Looking at the 2 options, I personally would rather go for the 1st option.

Reply:
Some time ago I believe we listened to the same socket for qc, gem2s and worker updates, so I believe the problem here is that gem2s and qc had their socket changed but the worker wasn't so it became broken. 
To solve this the best idea is to bring it back in line with the original solution.